src/python/system/new_process.py

```diff
@@ -376,3 +377,35 @@ def run_and_wait(self, *args, **kwargs):  # pylint: disable=arguments-differ
 
 class UnicodeProcessRunner(UnicodeProcessRunnerMixin, ProcessRunner):
   """ProcessRunner which always returns unicode output."""
+
+
+class UnshareProcessRunnerMixin(object):
+  """ProcessRunner mixin which unshares."""
+
+  def get_command(self, additional_args=None):
+    """Overridden get_command."""
+    if environment.platform() != 'LINUX':
+      raise RuntimeError('UnshareProcessRunner only supported on Linux')
+
+    unshare_path = spawn.find_executable('unshare')
+    if not unshare_path:
+      raise RuntimeError('unshare not found')
+
+    command = [unshare_path]
+    command.extend([
+        '-U',  # Enter user namespace.
+        '--map-root-user',  # Map user to root user.
+        '-n',  # Enter network namespace.
+    ])
```

Comment:
Does the permission propogate to child processes ?

Reply:
Yep!